geonode/upload/celery_tasks.py

```diff
@@ -80,6 +82,115 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         evaluate_error(self, exc, task_id, args, kwargs, einfo)
 
 
+class UpdateTaskClass(Task):
+    max_retries = 3
+    track_started = True
+    # We need a flag to check if the current task is the import_resource
+    # since it handles all the layers at the same time
+    bulk: bool = False
+
+    def on_success(self, retval, task_id, args, kwargs):
+        """
+        Called when the task succeeds.
+        Updates tasks_status for all alternates.
+        """
+        task_name = self.name
+        execution_id = args[0]
+        layer_key = find_key_recursively(kwargs, "layer_key")
+
+        self.set_task_status(task_name, execution_id, layer_key, "SUCCESS")
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        """
+        Called when the task fails.
+        Updates the ExecutionRequest.tasks dict and delegates logging/error handling to evaluate_error.
+        """
+        task_name = self.name
+        execution_id = args[0]
+        layer_key = find_key_recursively(kwargs, "layer_key")
+
+        self.set_task_status(task_name, execution_id, layer_key, "FAILED")
+
+        # Delegate the rest (errors, failed_layers, status) to evaluate_error
+        evaluate_error(self, exc, task_id, args, kwargs, einfo)
+
+    def before_start(self, task_id, args, kwargs):
+        """
+        Called before the task runs.
+        Marks the task as RUNNING for all alternates (or placeholder if none).
+        """
+        execution_id = args[0]
+        task_name = self.name
+
+        _exec = orchestrator.get_execution_object(execution_id)
+        tasks_status = _exec.tasks or {}
+
+        # If no layer exist yet (import task), use the placeholder
+        if not tasks_status:
+            tasks_status["pending_layer"] = {}
+
+        for layer_key, status_dict in tasks_status.items():
+            if status_dict.get(task_name) is None:
+                status_dict[task_name] = "PENDING"
+
+            orchestrator.update_execution_request_status(
+                execution_id,
+                tasks=tasks_status,
+            )
+
+    def set_task_status(self, task_name, execution_id, layer_key, status):
+        """
+        Set the task status for the on_success and on_failure celery methods
+        """
+        _exec = orchestrator.get_execution_object(execution_id)
```

Comment:
@Gpetrak as a note for the future, we should evaluate to have a cache for this function.